src/bsoncxx/test/oid.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iomanip>
+#include <sstream>
+
+#include <bsoncxx/oid.hpp>
+#include <bsoncxx/test_util/catch.hh>
+
+using namespace bsoncxx;
+
+namespace {
+
+void compare_string(const std::time_t& t, std::string time) {
+    std::stringstream time_stream;
+    time_stream << std::put_time(std::gmtime(&t), "%b %e, %Y %H:%M:%S UTC");
+    REQUIRE(time_stream.str() == time);
+}
+
+TEST_CASE("oid", "[bsoncxx::oid]") {
+    SECTION(
+        "represents the Timestamp field as an unsigned 32-bit representing the number of seconds "
+        "since the Epoch") {
+        // 0x00000000: To match "Jan 1st, 1970 00:00:00 UTC"
+        oid a{"000000000000000000000000"};
+        auto t = a.get_time_t();
+        REQUIRE(t == 0x00000000);
+        compare_string(t, "Jan  1, 1970 00:00:00 UTC");
+
+        // 0x7FFFFFFF: To match "Jan 19th, 2038 03:14:07 UTC"
+        oid b{"7FFFFFFF0000000000000000"};
+        t = b.get_time_t();
+        REQUIRE(t == 0x7FFFFFFF);
+        compare_string(t, "Jan 19, 2038 03:14:07 UTC");
+
+        // 0x80000000: To match "Jan 19th, 2038 03:14:08 UTC"
+        oid c{"800000000000000000000000"};
+        t = c.get_time_t();
+        REQUIRE(t == 0x80000000);
+        compare_string(t, "Jan 19, 2038 03:14:08 UTC");
+
+        // 0xFFFFFFFF: To match "Feb 7th, 2106 06:28:15 UTC"
+        oid d{"FFFFFFFF0000000000000000"};
+        t = d.get_time_t();
+        REQUIRE(t == 0xFFFFFFFF);
+        compare_string(t, "Feb  7, 2106 06:28:15 UTC");
+    }
+
+    SECTION("overflows predictably") {
+        /* The C++ driver does not have the ability to set a context for oids, so we
+           rely on the C driver's testing of this behaviour */
+    }
+
+    SECTION("oid", "uses a different machine/process number after fork is called") {
+        /* The C++ driver does not have the ability to set a context for oids, so we
+           rely on the C driver's testing of this behaviour */
```

Comment:
If this uses the default behavior, then I think we could test this. The default `bson_context_t` has the PID cache disabled: http://mongoc.org/libbson/current/bson_context_get_default.html

Reply:
Unfortunately the catch test framework does not support forking *sigh*

Reply:
Ah gotcha.